sdk/formrecognizer/Azure.AI.FormRecognizer/perf/Azure.AI.FormRecognizer.Perf/Scenarios/RecognizeCustomFormsWithoutLabels.cs

```diff
@@ -10,32 +10,35 @@
 
 namespace Azure.AI.FormRecognizer.Perf
 {
-    public class RecognizeCustomFormsWithoutLabelsTest : FormRecognizerTest<PerfOptions>
+    public sealed class RecognizeCustomFormsWithoutLabels: FormRecognizerTest<PerfOptions>
     {
         private readonly FormRecognizerClient _client;
 
         private readonly FormTrainingClient _trainingClient;
 
         private string _modelId;
 
-        public RecognizeCustomFormsWithoutLabelsTest(PerfOptions options) : base(options)
+        public RecognizeCustomFormsWithoutLabels(PerfOptions options) : base(options)
         {
-            _client = new FormRecognizerClient(new Uri(Endpoint), new AzureKeyCredential(ApiKey));
-            _trainingClient = new FormTrainingClient(new Uri(Endpoint), new AzureKeyCredential(ApiKey));
+            _client = new FormRecognizerClient(new Uri(TestEnvironment.Endpoint), new AzureKeyCredential(TestEnvironment.ApiKey));
+            _trainingClient = new FormTrainingClient(new Uri(TestEnvironment.Endpoint), new AzureKeyCredential(TestEnvironment.ApiKey));
         }
 
-        public override async Task SetupAsync()
+        public override async Task GlobalSetupAsync()
```

Comment:
Other changes are structural and look good.
Want to call out that this change `SetupAsync()` to `GlobalSetupAsync()` is functional. Earlier, the code would run for each test instance. Now, it will run exactly once (in the context of the first test instance to be precise, but that should not matter). I think that's what you want. Train once and then all test clients hit the same endpoint.

Reply:
That's what I want indeed. I had misconceptions about the use of `SetupAsync` and `GlobalSetupAsync` before, but the guide made it more clear.